modules/l4proxyprotocol/handler_test.go

```diff
@@ -0,0 +1,145 @@
+package l4proxyprotocol
+
+import (
+	"bytes"
+	"context"
+	"github.com/caddyserver/caddy/v2"
+	"github.com/mholt/caddy-l4/layer4"
+	"io"
+	"net"
+	"reflect"
+	"sync"
+	"testing"
+	"unsafe"
+)
+
+func assertString(t *testing.T, expected string, value string) {
+	t.Helper()
+	if value != expected {
+		t.Fatalf("Expected '%s' but got '%s'\n", expected, value)
+	}
+}
+
+func injectConfig(ctx *caddy.Context, cfg *caddy.Config) {
+	mutable := reflect.ValueOf(ctx).Elem()
+	field := mutable.FieldByName("cfg")
+	unsafePtrToField := unsafe.Pointer(field.UnsafeAddr())
+	ptrToField := (**caddy.Config)(unsafePtrToField)
+	*ptrToField = cfg
+}
```

Comment:
As mentioned in the issue, I will see what I can do to make it so this is not necessary -- yikes! :)

Reply:
I removed it with https://github.com/mholt/caddy-l4/pull/58/commits/51c94ed5d159160f663a49c40727c1140800c65c and instead increased the version of caddy in go.mod